src/LiveStream.js

```diff
@@ -1216,7 +1217,179 @@ class EluvioLiveStream {
 
   }
 
+  /**
+   * Calculate live streaming latency - ingest, egress, metadata.
+   */
+  async LatencyCalculator({status}) {
+
+    const debug = this.debug;
+    let stats = {};
+
+    if (debug) console.log("Latency calculator ", status.object_id);
+
+    const getMetaStart = process.hrtime();
+    let edgeMeta = await this.client.ContentObjectMetadata({
+      libraryId: status.library_id,
+      objectId: status.object_id,
+      writeToken: status.edge_write_token
+    });
+    const getMetaElapsed = process.hrtime(getMetaStart);
+    stats.meta_delay = (getMetaElapsed[0] * 1000000000 + getMetaElapsed[1]) / 1000000;
+
+    let recordings = edgeMeta.live_recording.recordings;
+    let sequence = recordings.recording_sequence;
+    let period = recordings.live_offering[sequence - 1];
+
+    let params = edgeMeta.live_recording.recording_config.recording_params;
+    let sourceTimescale = params.source_timescale;
+    let videoSegDurationTs = params.xc_params.video_seg_duration_ts;
+    let mezDurationMillis = videoSegDurationTs * 1000 / sourceTimescale;
+    let segDurationMillis = mezDurationMillis / 15;
+    let startTimeMillis = period.start_time_epoch_sec * 1000;
```

Comment:
In order to make this calculation accurate we need some changes in fabric:
- StartTimeEpochSec is based on seconds and not milli or micro. We need to change this to more accurate time measurement.
- We have not measured how long part finalization takes in the fabric and in latency calculator you assumed part finalization is zero (it is part of the delay).

Because of the above we can see errors up to 1 sec or maybe more (depending on part finalization time) in calculation.